crates/neon-runtime/src/napi/tsfn.rs

```diff
@@ -34,6 +35,7 @@ unsafe impl Sync for Tsfn {}
 /// function for scheduling tasks to execute on a JavaScript thread.
 pub struct ThreadsafeFunction<T> {
     tsfn: Tsfn,
+    is_finalized: Arc<Mutex<bool>>,
```

Comment:
Can this be an [`AtomicBool`](https://doc.rust-lang.org/std/sync/atomic/struct.AtomicBool.html) instead of a `Mutex<bool>`? I'm also not clear why it needs to be an `Arc`.

Reply:
`Arc` provides a nice way of passing the pointer to the C function, because unlike `Box` you can clone it before turning into a raw pointer, and then decrement the reference count when turning it back from the raw pointer to the `Arc` instance and dropping it.

I think it could be `AtomicBool`. Let me try that in a moment.

Reply:
Moved to `AtomicBool`. @jrose-signal should be really happy to see this happen :wink:

Reply:
Yeah, it has to be *something* indirect because ThreadsafeFunction is passed around by value. Box is *probably* safe because the finalizer callback is guaranteed to be invoked before `Drop::drop` completes (either Node invokes it on shutdown or we invoke it in `drop`), but it's a lot easier to mess that up somehow.